lib/jbuilder/jbuilder_template.rb

```diff
@@ -242,13 +244,19 @@ def _set_inline_partial(name, object, options)
     value = if object.nil?
       []
     elsif _is_collection?(object)
-      _scope{ _render_partial_with_options options.merge(collection: object) }
+      _scope do
+        options[:collection] = object
+        _render_partial_with_options options
+      end
     else
       locals = ::Hash[options[:as], object]
-      _scope{ _render_partial_with_options options.merge(locals: locals) }
+      _scope do
+        options[:locals] = locals
+        _render_partial_with_options options
+      end
     end
 
-    set! name, value
+    _set_value name, value
```

Comment:
We can set the value directly here instead of going back in through `set!` with different options. A call to `set!` with this options will just end up calling `_set_value` anyways.

This saves a bit in processing and also avoids an extra memory allocation for `*args`.